.evergreen/config.yml

```diff
@@ -82,11 +82,60 @@ functions:
         command: |
           astrolabevenv/${PYTHON_BIN_DIR}/astrolabe spec-tests run-one tests/${TEST_NAME}.yaml -e integrations/${DRIVER_DIRNAME}/workload-executor
 
+  "validate executor":
+    # Run a MongoDB instance locally.
+    - command: shell.exec
+      params:
+        working_dir: astrolabe-src
+        script: |
+          set -o xtrace
+
+          # User configurable-options
+          export MONGODB_VERSION="4.2"
+          export TOPOLOGY="replica_set"
+
+          # Setup variables
+          export DRIVERS_TOOLS="$(dirname $(pwd))/drivers-tools"
+          if [ "Windows_NT" = "$OS" ]; then
+             export DRIVERS_TOOLS=$(cygpath -m $DRIVERS_TOOLS)
+          fi
+          export MONGODB_BINARIES="$DRIVERS_TOOLS/mongodb/bin"
+          export MONGO_ORCHESTRATION_HOME="$DRIVERS_TOOLS/.evergreen/orchestration"
+          export PATH="$MONGODB_BINARIES:$PATH"
+
+          # Clone drivers-evergreen-tools
+          git clone --recursive git://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
+
+          # Configure mongo-orchestration
+          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
+
+          # Fix absolute paths
+          for filename in $(find ${DRIVERS_TOOLS} -name \*.json); do
+            perl -p -i -e "s|ABSOLUTE_PATH_REPLACEMENT_TOKEN|${DRIVERS_TOOLS}|g" $filename
+          done
+
+          # Make files executable
+          for i in $(find ${DRIVERS_TOOLS}/.evergreen -name \*.sh); do
+            chmod +x $i
+          done
+
+          # Run mongo-orchestration
+          sh $DRIVERS_TOOLS/.evergreen/run-orchestration.sh
+    # Validate the workload executor against the local MongoDB instance.
+    - command: subprocess.exec
+      type: test
+      params:
+        working_dir: astrolabe-src
+        add_expansions_to_env: true
+        command: |
+          astrolabevenv/${PYTHON_BIN_DIR}/astrolabe spec-tests validate-workload-executor -e integrations/${DRIVER_DIRNAME}/workload-executor --connection-string "mongodb://localhost:27017/?replicaSet=repl0"
+
   "delete test cluster":
     # Delete the cluster that was used to run the test.
     - command: subprocess.exec
       params:
         working_dir: astrolabe-src
+        continue_on_err: true       # Because we may not be running an actual spec test.
```

Comment:
this feels like a kludge to me, we shouldn't be cleaning up a cluster if one was never created (for the validation task)

Reply:
Removed the `continue_on_err`. Its not required as failures in `post` are ignored anyway.